License

jburel opened the pull request:
# What this PR does

 * Update the README with list of dependencies
 * Remove dependencies in insight

# Testing this PR

Check that the build is green

cc @chris-allan @jrswedlow

jburel committed:
Remove dependencies

jburel committed:
Update readme with dependencies for insight

jburel committed:
Add missing license files

jburel committed:
Add license files

jburel committed:
Add new entries

jburel committed:
Add new license file

jburel committed:
Add entries for python section

jburel committed:
Clean up

jburel commented:
--exclude

mtbc commented on lib/licenses/README:
```diff
@@ -329,10 +577,54 @@
+  Artifacts:    native-lib-loader-*.jar
+  Project:      A library for loading native libraries.
+  Web Site:     http://github.com/scijava/native-lib-loader
+  License:      Free BSD License
```
usually "FreeBSD" single word

mtbc commented on lib/licenses/README:
```diff
@@ -445,26 +776,66 @@
+                critical XML standards interfaces to make the lives of both our developers 
+                and users easier. The External Components portion of xml-commons contains 
+                interfaces that are defined by external standards organizations. For DOM, 
+                that's the W3C;
```
odd to end on a semicolon

mtbc commented:
We seem to have a couple of libraries with no license noted. Have we at least a Trello card for investigating further?

jburel commented:
@mtbc @will-moore is looking at the ones I left

will-moore commented:
SpinBtn is long dead. No license info and the only url I can find is 404 http://www.softwareunity.com/jquery/JQuerySpinBtn/
Most jQuery stuff uses MIT license

jburel committed:
Re-adding dependencies

They should probably need to be moved to another ivy.xml but this
can be done in another PR.

jburel commented:
I will add to  https://trello.com/c/zNVVhCR2/28-javascript-and-python-dependencies-review so we can look at removing it

will-moore commented on lib/licenses/README:
```diff
@@ -564,162 +961,241 @@
-  Web Site:     http://raphaeljs.com/
-  License:      MIT license.
-  License File: jquery.txt
+  Artifacts:    components/tools/OmeroWeb/omeroweb/webgateway/static/3rdparty/ie8_pngfix/*
```
I can't see that we use this anywhere.

will-moore commented on lib/licenses/README:
```diff
@@ -564,162 +961,241 @@
+  Artifacts:    components/tools/OmeroWeb/omeroweb/webgateway/static/3rdparty/jquery.selectboxes-2.2.6.js
+  Web Site:     
+  License:      MIT License or GNU GENERAL PUBLIC LICENSE
+  License File: mit.txt, gpl-v3.0.txt
```
Can't see that we use this anywhere.

will-moore commented on lib/licenses/README:
```diff
@@ -564,162 +961,241 @@
-  Web Site:     http://underscorejs.org/
-  License:      BSD
-  License File: bsd.txt
+  Artifacts:    components/tools/OmeroWeb/omeroweb/webgateway/static/3rdparty/aops-1.3.js
```
We don't use this aops-1.3.js library anywhere.

jburel committed:
Add axis entries

will-moore commented on lib/licenses/README:
```diff
@@ -564,162 +961,241 @@
+  License File: mit.txt
+
+
+  Artifacts:    components/tools/OmeroWeb/omeroweb/webgateway/static/3rdparty/pie-1.0beta5/*
```
I'm sure we can remove this now - only for old IE.

jburel committed:
Minor edits

jburel committed:
Add missing section

jburel commented:
@will-moore I will not remove them from the code in the point release. 
I have added them to the card mentioned above.
They have similar licenses that some already listed

sbesson commented:
@jburel: one thought while scanning `lib/licenses/README` is that we are not listing the Bio-Formats components (`formats-gpl`...). Is that a deliberate choice for our own dependencies?

jburel commented:
I did not list any of our own dependencies.
Happy to do so if we think this is required

chris-allan commented:
I would think it safe to link to the Bio-Formats dependencies rather than potentially introduce copy and paste errors and additional maintenance burden by also including them in `lib/licenses/README`.

sbesson commented:
As we are independently maintaining the list Bio-Formats dependencies, I am definitely in favor of linking rather than trying to maintain in two places. I will review this list and flag all the low-level dependencies worth discussing.

jburel committed:
Add B-F dependencies

sbesson commented on lib/licenses/README:
```diff
@@ -658,6 +665,47 @@
+
+
+  Artifacts:    ome-xml-*.jar
+  Project:      Java API to handle Microsoft MDB format (Access).
```
That's the same description as `ome-mdbtools`

sbesson reviewed (commented):
Compared the content of the license README with the dependency report generated by Ivy and bundled under `share/{server,client}` in OMERO.server. 

The main missing dependencies I could find are:

- `formats-api`, `formats-gpl`, `formats-bsd`
- `bcprov-jdk14`, `bcmail-jdk14`
- `jts`
- `jaxrpc-api`
- `axis-saaj` should be `axis2-saaj`

jburel committed:
Add new library file

jburel committed:
Add missing entries

jburel commented:
@sbesson the list should now be complete

sbesson commented:
The last commit addresses my comments from https://github.com/openmicroscopy/openmicroscopy/pull/5335#pullrequestreview-45710189
